### ISSUE TITLE
mergeReferences config

### ISSUE THREAD

**Issue by fritzschoff** [dappsnation/akita-ng-fire#126]
```
Ok I got the issue.
It's happened when I try to add reference field in document.


My firebase settings was OK.

Updated Sample: https://stackblitz.com/edit/angular-fvaf5r

Please check and let me know. :)
```

**Comment by fritzschoff** [comment]
@saifxx so did a bit of research and a question popped in my mind. How do you think this document reference should be represented in akita? Cause what you get in the is just a simple [document reference ](https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentReference). You can query that by yourself in the [formatFromFirestore](https://github.com/dappsnation/akita-ng-fire/blob/v3.0.0/projects/akita-ng-fire/src/lib/collection/collection.service.ts#L122) hook. Depending on your store in akita, may it be a plain store or an entity store, you can parse the data as you want. I don't think it would be a good idea to force a behavior like creating a new key in the store with the id of the document with the values as the properties. 
I can make these hooks async so you can `await` them.

**Comment by saifurrehmanx** [comment]
Yes you are right, we should not change document reference. But there is no other way than converting document reference to the id of the document.

Or maybe you can warn user in documentation that "You must have to change document reference to the id of the document in formatFromFirestore."

In my previous project, I'm already handling this situation in _fromFirestore_ by creating a new key with the id of the document with the values as the properties:

![image](https://user-images.githubusercontent.com/29164221/87019241-cbddaf80-c1eb-11ea-8034-8de5b08ccef1.png)

**Comment by fritzschoff** [comment]
I guess there are more then one way.
- Maybe you want to update a new store (no entitiy store).
- Maybe you want to update another enitityStore with the docRef.

For me, it seems the best way to let the user decide how to handle that docRef in the formatFormFirestore function. He can then run his own `runEntityStoreAction`  or  `runStoreAction`.
Want you might can have, is a flag in the `CollectionConfig` that says if  docRef is present, merge it in the store like you did in the screenshot. 
`@CollectionConfig({ path: 'prodcuts', mergeReferences: true })`

**Comment by saifurrehmanx** [comment]
Yes, that will be the best way.

**Comment by fritzschoff** [comment]
you mind doing a PR if you really need it? ATM I'm a bit short on time and not sure when I can do it.

**Comment by saifurrehmanx** [comment]
I'm interested to do PR, but I have never did this before. I don't know about GitHub too much. I will search on google for that. :)

**Comment by fritzschoff** [comment]
clone the repo, checkout branch v3.0.0 and open up a branch from this branch and you should be good to go.

**Comment by fritzschoff** [comment]
@saifxx you currently working on that?

**Comment by saifurrehmanx** [comment]
@fritzschoff I'm not working on it.

**Comment by fritzschoff** [comment]
@saifxx  this feature will be available in 4.1.1
